secret_handlers_test.go

```diff
@@ -879,3 +879,147 @@ func TestSecretEditBadRequest(t *testing.T) {
 
 	templates.BadRequestTest(t, rec)
 }
+
+func TestSecretDelete(t *testing.T) {
+	testutils.SkipIfNotRunningIntegrationTests(t)
+
+	util.OverrideEncryptionKey(strings.Repeat("test", 8))
+
+	tenantIDForSecret := int64(1)
+	testUserId := "testUser"
+
+	userDao := dao.GetUserDao(&tenantIDForSecret)
+	user, err := userDao.FindOrCreate(testUserId)
+	if err != nil {
+		t.Error(err)
+	}
+
+	secretExtra := map[string]interface{}{"extra": "params"}
+	password := "test"
+
+	requestBody := m.SecretEditRequest{
+		Extra:    &secretExtra,
+		Password: util.StringRef(password),
+	}
+
+	body, err := json.Marshal(requestBody)
+	if err != nil {
+		t.Error("Could not marshal JSON")
+	}
+
+	var userID *int64
+	for _, userScoped := range []bool{false, true} {
+		if userScoped {
+			userID = &user.Id
+		}
+
+		secret, err := dao.CreateSecretByName("Secret 1", &tenantIDForSecret, userID)
+		if err != nil {
+			t.Error(err)
+		}
+
+		secretID, err := util.InterfaceToString(secret.DbID)
+		if err != nil {
+			t.Error(err)
+		}
+
+		c, rec := request.CreateTestContext(
+			http.MethodDelete,
+			"/api/sources/v3.1/secrets/"+secretID,
+			bytes.NewReader(body),
+			map[string]interface{}{
+				"tenantID": tenantIDForSecret,
+			},
+		)
+
+		c.SetParamNames("id")
+		c.SetParamValues(secretID)
+		c.Request().Header.Add("Content-Type", "application/json;charset=utf-8")
+
+		tenancy := middleware.Tenancy(SecretDelete)
+		userCatcher := middleware.UserCatcher(tenancy)
+		identityHeader := testutils.IdentityHeaderForUser(testUserId)
+		c.Set("identity", identityHeader)
+
+		err = userCatcher(c)
+		if err != nil {
+			t.Error(err)
+		}
+
+		if rec.Code != http.StatusNoContent {
+			t.Error("Did not return 204")
+		}
+
+		secretDao := dao.GetSecretDao(&dao.RequestParams{TenantID: &tenantIDForSecret, UserID: userID})
+		_, err = secretDao.GetById(&secret.DbID)
+		if err.Error() != "secret not found" {
+			t.Error("'secret not found' expected")
+		}
```

Comment:
maybe here we can check if returned err is the type we want instead of compare the strings, something like: 
```
		if !errors.Is(err, util.ErrNotFoundEmpty) {
			t.Error("'secret not found' expected")
		}
```